crates/ruff_cli/build.rs

```diff
@@ -1,3 +1,43 @@
-fn main() -> shadow_rs::SdResult<()> {
-    shadow_rs::new()
+use std::process::Command;
+
+fn main() {
+    commit_info();
+    #[allow(clippy::disallowed_methods)]
+    let target = std::env::var("TARGET").unwrap();
+    println!("cargo:rustc-env=RUST_HOST_TARGET={target}");
+}
+
+fn commit_info() {
+    let output = match Command::new("git")
```

Comment:
How does this work caching wise, i.e. how do we avoid cargo using a cache old hash? I think the only solution is https://doc.rust-lang.org/cargo/reference/build-scripts.html#rerun-if-changed with `.git/HEAD`, which is also slightly annoying because it invalidates the cache everytime you do something with git.

I'd prefer to read files over subcommands (they have an overhead and weird os failure modes), but that might be to hard with git.

Reply:
I think it should be fine for release builds. For local builds, yes it may be incorrect unless you `cargo clean`. I'm not sure if it's worth adding the rerun-if-changed?

I've only ever seen git subcommand used for this, it definitely seems hard to read files.

Reply:
We could use https://doc.rust-lang.org/cargo/reference/build-scripts.html#rerun-if-env-changed and set an environment variable in CI on release builds for safety? Could be used to force a refresh locally too.

Reply:
That sounds like a good workaround

Reply:
`.git/HEAD` only contains a pointer to the current branch so I don't think that would work (unless git sets a new mtime on the file on each operation)

> Normally build scripts are re-run if any file inside the crate root changes, but this can be used to scope changes to just a small set of files.

It seems like we'd only be reducing scope by doing this?

Unfortunately the environment variable one is not "if set" it's "if changes" so it'd work well in CI but not so much for local resets.

Reply:
Addressed in https://github.com/astral-sh/ruff/pull/8034/commits/1208df9f40687c7a58aec301723919bd15087d3e